[Docs] Add CSS customization guide (#716)
mckinsey/vizro: vizro-core/docs/pages/user-guides/custom-css.md

# Customizing Vizro dashboard CSS

> Vizro is opinionated about visual formatting, but it allows for some customization if you want to

Reviewer: ```suggestion
Vizro is opinionated about visual formatting, and some elements, such as the layout of the navigation and components, are fixed. You can customize some settings such as
```

Reviewer: Suggested change to the intro to say first what is opinionated and fixed, and then what you can change. You can leave what you have if it's simpler though!

> deviate from the default settings. While certain elements, such as the layout of the navigation and components,

Reviewer: Remove this

> are fixed, you can customize background colors, fonts, and other styles via CSS overrides.

Reviewer: And remove this

> To do so, you need to:

Reviewer: ```suggestion
To make customizations, you need to:
```

> 1. Add a CSS file to your `assets` folder, refer to our user guide on [adding static assets](assets.md#how-to-add-static-assets).

Reviewer: ```suggestion
1. **Add a CSS file to your `assets` folder**. Refer to our user guide on [adding static assets](assets.md#how-to-add-static-assets).
```

> 2. Identify the correct CSS selector for the component you want to style.

Reviewer: ```suggestion
2. **Identify the correct CSS selector** for the component you want to style.
```

> 3. Add the desired changed properties to your CSS file.

Reviewer: ```suggestion
3. **Add the desired change** to the properties to your CSS file.
```

> ## Introduction to CSS

Author: This section is less about Vizro CSS, but gives a general intro into CSS. Shall I still rename it for better google finds? 🤔

> ## Introduction to CSS

Reviewer: Yes, keep it with Vizro in it to help it rank for queries specifically about Vizro

> Customizations to the visual formatting of a Vizro dashboard is typically done using CSS. For a short introduction to

Reviewer: ```suggestion
Customizations to the visual formatting of a Vizro dashboard are typically done using CSS. For a short introduction to
```

HTML document of your Vizro app.

    ![Inspect panel](../../assets/user_guides/custom_css/inspect-panel.png)

2. **Select an Element:** Suppose you want to change the background color of your cards. Click the
"Select an element in the page to inspect it" icon in the top left corner of the inspect panel.

    ![Inspect icon](../../assets/user_guides/custom_css/inspect-icon.png)

3. **Find the HTML Block:** Hover over the component you want to style. The corresponding HTML block will be
highlighted in the document.

    ![Highlighted element](../../assets/user_guides/custom_css/highlighted-element.png)

    Notice that the selected HTML block corresponds to the container of the card and has a CSS class, here it is `card`.

4. **Apply CSS:** Use this CSS class to style the card component. In your CSS file, you can write:

    ```
    .card {
        background-color: blue;
    }
    ```

    This changes the background color for any HTML element with the `card` class.

**Tip:** You can also test your CSS live by editing the CSS attributes in the "Elements" panel.

> For example, temporarily add `background: blue; note that this change will be lost upon reloading the page.

Reviewer: ```suggestion
For example, temporarily add `background: blue`; note that this change will be lost upon reloading the page.
```

        import vizro.models as vm
        import vizro.plotly.express as px
        from vizro import Vizro

        iris = px.data.iris()

        page = vm.Page(
            title="Page with subsections",
            layout=vm.Layout(grid=[[0, 1]]),
            components=[
                vm.Container(
                    title="Container I",
                    components=[vm.Graph(figure=px.scatter(iris, x="sepal_width", y="sepal_length", color="species"))]
                ),
                vm.Container(
                    title="Container II",
                    components=[vm.Graph(figure=px.box(iris, x="species", y="sepal_length", color="species"))]
                )
            ],
        )

        dashboard = vm.Dashboard(pages=[page])
        Vizro().build(dashboard).run()
        ```

        <img src=https://py.cafe/logo.png alt="py.cafe logo" width="30"><b><a target="_blank" href="https://py.cafe/vizro-official/vizro-style-a-container">Run and edit this code in Py.Cafe</a></b>

    === "app.yaml"
        ```yaml
        # Still requires a .py to add data to the data manager and parse YAML configuration
        # See yaml_version example
        pages:

>         - components:

Author: Yaml example to be updated
